test/framework/src/test/java/org/elasticsearch/test/AbstractXContentTestCaseTests.java

```diff
@@ -0,0 +1,48 @@
+package org.elasticsearch.test;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class AbstractXContentTestCaseTests extends ESTestCase {
+
+    public void testInsertRandomFieldsAndShuffle() throws IOException {
+        TestInstance t = new TestInstance();
+        boolean randomFieldAtFirstPosition = false;
+        for (int i = 0; i < 5; i++) {
```

Comment:
Five repetitions is unfortunately not enough to always hit a case where the randomization kicks in. I ran the test with 100 repetitions and it failed 2 times out of that. Too much for out CI. I also wouldn't say the number should be increased. However, it would be great to "fix" the randomness in this particular case, since we are testing a test, but want it to behave in a specific way (do the shuffling). I did some digging in LuceneTestCase and fortunately found this little gem: RandomizedContext#runWithPrivateRandomness(). This seems to overwrite the test suite source of randomness for the time of invocation of a Callable, which in our case would be enough to make the shuffling always have some effect. We could fix the seed like this and avoid the loop alltogether:

```
        // we need to "fix" randomness in order to be sure the shuffling takes effect
        BytesReference insertRandomFieldsAndShuffle = RandomizedContext.current().runWithPrivateRandomness(1,
                () -> AbstractXContentTestCase.insertRandomFieldsAndShuffle(t, XContentType.JSON, true, new String[] {}, null,
                        this::createParser, ToXContent.EMPTY_PARAMS));
```

Note that with differnt seed (like e.g. 0) the test consistently fails, so we should use a "good" one.
I haven't used this before and also would like the opinion of @javanna if this makes sense. I would much prefer this to increasing the current loop count, which also doesn't really makes certain the test succeeds.

Reply:
sounds good to me

Reply:
@cbuescher thanks for digging into this ;) 
I did not know of the `runWithPrivateRandomness` which works great in this case !